src/commands/timeout.ts

```diff
@@ -0,0 +1,162 @@
+
+        if (interaction.inGuild() === false) {
+            return interaction.reply({ content: "❌ I can't execute this command inside DMs!", ephemeral: true });
+        }
+
+        if (!interaction.guild.members.me.permissions.has(PermissionFlagsBits.ModerateMembers)) {
+            return interaction.reply({
+                content: '❌ Sorry, I need the `Timeout Members` in order to execute this command.',
+                ephemeral: true,
+            });
+        }
+        if (typeof interaction.member.permissions === 'string')
+            return interaction.reply({ content: '❌ Unknown permissions. Please try again later.', ephemeral: true });
+        else if (!interaction.member.permissions.has(PermissionFlagsBits.ModerateMembers)) {
+            return interaction.reply({
+                content: '⛔ You need the `Timeout Members` permission in order to use this command!',
+                ephemeral: true,
+            });
+        }
+
+        const user = interaction.options.getUser('user', true);
+        const member = await interaction.guild.members.fetch(user.id);
+
+        if (!member) {
+            return interaction.reply({
+                content: '❌ You need to specify a valid user to manage their timeout!',
+                ephemeral: true,
+            });
+        }
+
+        switch (interaction.options.getSubcommand()) {
+            default:
+            case 'add': {
+                if (member.id === interaction.user.id) {
+                    return interaction.reply({ content: "Aww, please don't time out yourself! 💖", ephemeral: true });
+                }
+
+                if (member.moderatable === false) {
+                    return interaction.reply({
+                        content:
+                            '❌ I cannot time out this user! \n**Please make sure that my highest role is above theirs.**',
+                        ephemeral: true,
+                    });
+                }
+
+                const duration = interaction.options.getString('duration', true);
+                const msDuration = ms(duration);
+
+                if (msDuration == null) {
+                    return interaction.reply({
+                        content: '❌ Please specify a valid timeout duration.',
+                        ephemeral: true,
+                    });
+                }
+
+                if (msDuration <= 0) {
+                    return interaction.reply({
+                        content: '❌ The duration of the timeout must be greater than 0!',
+                        ephemeral: true,
+                    });
+                }
+
+                let response = `✅ Timed out \`${member.user.tag} (${member.id})\`.`;
+
+                if (!interaction.options.getBoolean('silent')) {
+                    const notified = await notifyUser(user, interaction, 'timed out');
+                    if (!notified) response += "\n\n⚠️ Couldn't send DM to user.";
+                }
```

Comment:
This gets sent even if the timeout itself fails. I was thinking maybe we could put it inside the try/catch statement, or below it?